openmls/src/group/tests/test_proposal_validation.rs

```diff
@@ -204,7 +423,84 @@ fn test_valsem101(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider
         }
     }
 
-    // TODO #525: Add test for incoming proposals.
+    // Before we can test reception of (invalid) proposals, we set up a new
```

Comment:
Can you describe first on a high level what you're testing here? Same for the other tests.

Reply:
I added a remark to each test, indicating that we're now testing the same validation logic on the receiver side. Is that what you meant?